components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/importexport/utils/APIExportUtil.java

```diff
@@ -215,6 +216,66 @@ private static void exportAPIThumbnail(String archivePath, APIIdentifier apiIden
         }
     }
 
+
+    /**
+     * Retrieve SOAP to REST mediation logic for the exporting API and store it in the archive directory
+     *
+     * @param apiIdentifier ID of the requesting API
+     * @param registry      Current tenant registry
+     * @throws APIImportExportException If an error occurs while retrieving image from the registry or
+     *                            storing in the archive directory
+     */
+    private static void exportSOAPToRESTMediation(String archivePath,APIIdentifier apiIdentifier, Registry registry)
+            throws APIImportExportException {
+        String soapToRestBaseUrl = "/apimgt/applicationdata/provider" + RegistryConstants.PATH_SEPARATOR +
+                apiIdentifier.getProviderName() + RegistryConstants.PATH_SEPARATOR +
+                apiIdentifier.getApiName() + RegistryConstants.PATH_SEPARATOR +
+                apiIdentifier.getVersion() + RegistryConstants.PATH_SEPARATOR +
+                "soap_to_rest";
+
+        InputStream inputStream = null;
+        OutputStream outputStream = null;
+        try {
+            if (registry.resourceExists(soapToRestBaseUrl)) {
+                Collection inFlow = (org.wso2.carbon.registry.api.Collection)registry.get(soapToRestBaseUrl
+                        + RegistryConstants.PATH_SEPARATOR + "in");
+                Collection outFlow = (org.wso2.carbon.registry.api.Collection)registry.get(soapToRestBaseUrl
+                        + RegistryConstants.PATH_SEPARATOR + "out");
+
+                CommonUtil.createDirectory(archivePath + File.separator + "SoapToRest/in");
+                CommonUtil.createDirectory(archivePath + File.separator + "SoapToRest/out");
+                if (inFlow != null) {
+                    for (String inFlowPath : inFlow.getChildren()) {
+                        inputStream = registry.get(inFlowPath).getContentStream();
+                        outputStream = new FileOutputStream(archivePath + File.separator + "SoapToRest"
```

Comment:
Shall we use constants for "in", "out", "SoapToRest" strings?

Reply:
resolved by #8332